compiler/noirc_frontend/src/elaborator/comptime.rs

```diff
@@ -653,4 +665,41 @@ impl<'context> Elaborator<'context> {
                 _ => false,
             }
     }
+
+    /// Pushes an ElaborateReason but also takes the current errors and returns them.
+    pub(crate) fn push_elaborate_reason(
```

Comment:
Please rename to make it clear that this resets `self.errors`

Reply:
Done!